executor/executor_test.go

```diff
@@ -4030,6 +4030,21 @@ func (s *testSuite3) TestDoSubquery(c *C) {
 	c.Assert(r, IsNil, Commentf("result of Do not empty"))
 }
 
+func (s *testSuite3) TestSubqueryTableAlias(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec(`use test`)
+	tk.MustExec(`drop table if exists t`)
+
+	tk.MustGetErrCode("select a, b from (select 1 a) ``, (select 2 b) ``;", mysql.ErrDerivedMustHaveAlias)
+	tk.MustGetErrCode("select a, b from (select 1 a) `x`, (select 2 b) `x`;", mysql.ErrNonuniqTable)
+	tk.MustGetErrCode("select a, b from (select 1 a), (select 2 b);", mysql.ErrDerivedMustHaveAlias)
+
+	tk.MustExec(`set sql_mode = 'oracle';`)
+	tk.MustExec("select a, b from (select 1 a) ``, (select 2 b) ``;")
+	tk.MustExec("select a, b from (select 1 a) `x`, (select 2 b) `x`;")
+	tk.MustExec("select a, b from (select 1 a), (select 2 b);")
+}
```

Comment:
1. please ensure all 3 tests in 4043–4045 returns `1 2` (use `MustQuery`) 
2. all of the following should still result in `ErrNonUniq` (ERROR 1052 (23000): Column 'a' in field list is ambiguous):

    ```sql
    select a from (select 1 a) ``, (select 2 a) ``;
    select a from (select 1 a) `x`, (select 2 a) `x`;
    select x.a from (select 1 a) `x`, (select 2 a) `x`;
    select a from (select 1 a), (select 2 a);
    ```